packages/core/admin/admin/src/components/CommandPalette/pages/ContentManager.js

```diff
@@ -0,0 +1,47 @@
+import React, { useMemo } from 'react';
+import { Pencil } from '@strapi/icons';
+import { useModels } from '../../../hooks';
+import Items from '../Items';
+import { useCommand } from '../context';
+
+const useContentTypesItems = () => {
```

Comment:
Avoiding to bloat the component function and separating concerns. But I'm fine not splitting if this is better for some reason

Reply:
I think lets keep it in the component pls 👍🏼

Reply:
Sure, can you explain how this is better in the meantime ?

Reply:
Sure, can you explain how this is better in the meantime ?

Reply:
IMO it's a smaller interface of the file, abstracting the useModal & memoization adds a cognitive overload that is unnecessary especially since we're not sharing this hook logic anywhere else, if it were being exported then of course it would need to be separate but imo due to it's simplicity having a deeper component is easier to understand 👍🏼 

This is obvs just my opinion

Reply:
Thank for sharing, I would disagree on the cognitive load but I guess it depends how you look at it.

That's not important anyway I'll leave it in one compo to stay consistent with how it's done elsewhere